messaging/src/main/java/org/axonframework/eventhandling/pooled/PooledStreamingEventProcessor.java

```diff
@@ -319,7 +319,9 @@ public <R> void resetTokens(@Nonnull TrackingToken startPosition, R resetContext
             // Update all tokens towards ReplayTokens
             IntStream.range(0, tokens.length)
                      .forEach(i -> tokenStore.storeToken(
-                             ReplayToken.createReplayToken(tokens[i], startPosition), getName(), segments[i]
+                             ReplayToken.createReplayToken(tokens[i], startPosition, resetContext),
```

Comment:
Nice to see how easy this addition is from the perspective of the Event Processor. :-)